jurity/recommenders/interlist_diversity.py

```diff
@@ -0,0 +1,214 @@
+        k: int
+            Top-k recommendations to consider.
+        user_id_column: str
+            User id column name.
+        item_id_column: str
+            Item id column name.
+        click_column: str
+            Recommendation score column name.
+        sample_size: float
+            Proportion of users to randomly sample for evaluation.
+        seed : int
+            The seed used to create random state.
+        chunk_size: int
+            Chunk size to limit memory usage.
+
+        Returns
+        -------
+        Inter-list diversity metric, number of unique users as the support to get the metric
+        """
+
+    # Sample users
+    if sample_size is not None:
+        df = sample_users(predicted_results, user_id_column, sample_size, seed)
+    else:
+        df = predicted_results
+
+    # Sort by user and score
+    df = df.sort_values(click_column, ascending=False).groupby(user_id_column).head(k)
+
+    # Given user/item id maps, create sparse matrix.
+    sparse_matrix = tocsr(df, user_id_column, item_id_column)
+
+    # Get pairwise cosine similarities
+    similarities_sum = 0
+    if chunk_size == 1:
+        num_chunks = sparse_matrix.shape[0]
+    else:
+        num_chunks = sparse_matrix.shape[0] // chunk_size + 1
+    for i in range(num_chunks):
+        similarities = cosine_similarity(sparse_matrix[i * chunk_size:(i + 1) * chunk_size], sparse_matrix,
+                                         dense_output=False)
+        similarities_sum += similarities.sum()
+    similarities_sum = (similarities_sum - sparse_matrix.shape[0]) / 2.0
+
+    # Get number of pairs
+    num_pairs = np.sum(range(sparse_matrix.shape[0]))
+
+    # Calculate metric
+    if num_pairs == 0:
+        inter_list_diversity = np.nan
+        warnings.warn('Inter-List Diversity will be nan when there is only one single user.')
+    else:
+        inter_list_diversity = 1.0 - similarities_sum / num_pairs
+        if np.abs(inter_list_diversity) <= 1e-06:
+            inter_list_diversity = 0.0
+
+    # Calculate support, set it to be the number of users
+    support = len(df[user_id_column].unique())
+
+    return inter_list_diversity, support
+
+
+class InterListDiversity:
+    """Inter-List Diversity@k
+
+    Inter-List Diversity@k measures the inter-list diversity of the recommendations when only k recommendations are
+    made to the user. It measures how user's lists of recommendations are different from each other.
+
+    .. math::
+            Inter-list diversity = 1 - average(cosine_similarity(R_{u_i}, R_{u_j}))
+
+    where :math:`R_{u_i}` is the binary indicator vector representing provided recommendations for user :math:`u_i` and
+    :math:`i<j`.
+
+    Sources: https://towardsdatascience.com/evaluation-metrics-for-recommender-systems-df56c6611093
+    """
+
+    def __init__(self, click_column, k: int = None, user_id_column: str = Constants.user_id,
```

Comment:
Can you add documentation to this function? Since there are non-standard parameters (seed, chunk_size), it would be good to define them here.